Add support for 10.12.4

gdisaac07 opened the pull request:
Beta 2 introduced for 10.12.4 and breaks software. Added the following to fix broken code:

md5_version = {
    "00e2f0eb5db157462a83e4de50583e33": ["10.12.1 (16B2659)"],
    "ebde660af1f51dc7482551e8d07c62fd": ["10.12.2 (16C67)"],
    "7fbc510cf91676739d638e1319d44e0e": ["10.12.3 (16D32)"],
    "ad1f6d9f26deb5fd74b40e672141d65b": ["10.12.4 Beta 1 (16E144)"],
    **"064635706e72e76d4e4b4bd11084304e": ["10.12.4 Beta 2 (16E154)"]**
}

md5_patch = {
    "00e2f0eb5db157462a83e4de50583e33": "a6c2143c2f085c2c104369d7a1adfe03",
    "ebde660af1f51dc7482551e8d07c62fd": "2ebb68137da4a1cb0dfc6e6f05be3db2",
    "7fbc510cf91676739d638e1319d44e0e": "0af475c26cdf5e26f8fd7e4341dadea5",
    "ad1f6d9f26deb5fd74b40e672141d65b": "99ab120bd6e959029f1f84c8f62a63c8",
    **"064635706e72e76d4e4b4bd11084304e": "0a65726a07b0afaaa9390c73cf60ccaa"**

    
}

re_md5 = {
    0: [
        "00e2f0eb5db157462a83e4de50583e33",
        "ebde660af1f51dc7482551e8d07c62fd",
        "7fbc510cf91676739d638e1319d44e0e",
        "ad1f6d9f26deb5fd74b40e672141d65b",
        **"064635706e72e76d4e4b4bd11084304e"**
        ],
}

KhaosT committed:
Update for TB3

Commit:
Update tb3-enabler.py to support macOS 10.12.2

    Manually patched my IOThunderboltFamily kext in 10.12.2, and computed the md5 before and after patching.

    Now my output of running the script is:
    ```
    target: patched, 10.12.2 (16C67)
    backup: original, 10.12.2 (16C67)
    ```

KhaosT committed:
Merge pull request #1 from IMcD23/patch-1

Update tb3-enabler.py to support macOS 10.12.2

KhaosT committed:
Add missing field for 10.12.2

Commit:
Update for macOS 10.12.3 16D32 (#4)

lloeki commented:
@gdisaac07 I'm not entirely sure you submitted this to the right place, or am I missing something?

gdisaac07 commented:
Sorry, I added to my comment after I initially posted it.

danieltwagner committed:
Update for macOS 10.12.4 16E195 (#5)

KhaosT committed:
Update for 10.12.5

haltiamreptar committed:
Added changes to make code Python 3 compatible. (#6)

TPNxl commented:
[IOThunderboltFamily kext and macOSDeveloperBetaAccessUtility dmg.zip](https://github.com/lloeki/trim_patcher/files/1055970/IOThunderboltFamily.kext.and.macOSDeveloperBetaAccessUtility.dmg.zip)
Hello,
Can you guys create a version for 10.13? I'm using a Razer Core with the TI82 controller and I need to use this to enable support for it.
Thanks,
T
PS: I've attached a profile to download the 10.13 dev beta and the IOThunderboltFamily.kext from 10.13.

KhaosT committed:
Add ability to force patch the binary.

breath-co2 committed:
add 10.12.6 (16G29) version (#8)

add for 10.12.6 (16G29) version md5 data.

schwabe committed:
Add support for OS X 10.13 (#10)

* Allow tb3 enabler to work on OS x 10.13

Mac OS X slightly changes the code in the IOThunderboltSwitchType3::shouldSkipEnumeration() method. This adds support for 10.13 in the force apply method

* Allow 10.13 to patched without using forceApply

marcorosa committed:
Fix disabled sip issue (#9)

In my 10.12.5 this script always exited at `check_SIP` because `sip_info` variable got `g%00%00%00`. I suggest to remove the `w`

schwabe committed:
Fix python3 support (closes #11) (#12)

jameswhickmott committed:
Update tb3-enabler.py (#13)

Added 10.13.1 support

Awarua- committed:
Add support for 10.12.6 (16G1036) (#14)

ahjsrhj committed:
Update for 10.13.2 (#15)

* Update for 10.12.6

* Update for 10.13.2

* add version

Awarua- committed:
update to latest version for 10.12.6 (16G1114) (#16)

Awarua- committed:
Update for 13.3 (#17)

* update to latest version for 10.12.6 (16G1114)

* update for 13.3

KhaosT committed:
Update README.md

likaci committed:
Add support for 10.12.6 (16G1212) (#18)

schwabe committed:
Add Mac OS 10.13.4 (17E199) (#19)

schwabe committed:
Add 10.13.4 (17E202) (#20)

gekowa committed:
Added 10.14 (18A391) (#21)

Commit:
Update to 10.14.2 (#22)

* Update for macOS 10.12.3 16D32

* Add 10.14.2 support.

feijisu27 committed:
- fix the 10.14.2 issue. (#23)

- uncomment the old key.

feijisu27 committed:
could you approve this fix for 10.14.5 (#25)

* - fix the 10.14.2 issue.
- uncomment the old key.

* fix 10.14.5 issue.
